components/brave_federated/data_stores/ad_notification_timing_data_store.cc

```diff
@@ -0,0 +1,139 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_federated/data_stores/ad_notification_timing_data_store.h"
+
+#include <string>
+#include <utility>
+
+#include "base/bind.h"
+#include "base/strings/stringprintf.h"
+#include "base/time/time.h"
+#include "brave/components/brave_federated/data_stores/data_store.h"
+#include "sql/recovery.h"
+#include "sql/statement.h"
+#include "sql/transaction.h"
+
+namespace {
+
+void BindSampleLogToStatement(
+    const brave_federated::AdNotificationTimingTaskLog& notification_timing_log,
+    sql::Statement* s) {
+  s->BindInt64(0, notification_timing_log.time.ToInternalValue());
+  s->BindString(1, notification_timing_log.locale);
+  s->BindInt(2, notification_timing_log.number_of_tabs);
+  s->BindBool(3, notification_timing_log.label);
+  s->BindInt64(4, notification_timing_log.creation_date.ToInternalValue());
+}
+
+}  // namespace
+
+namespace brave_federated {
+
+// AdNotificationTimingTaskLog ---------------------------------------
+
+AdNotificationTimingTaskLog::AdNotificationTimingTaskLog(
+    const int id,
+    const base::Time& time,
+    const std::string& locale,
+    const int number_of_tabs,
+    const bool label,
+    const base::Time& creation_date)
+    : id(id),
+      time(time),
+      locale(locale),
+      number_of_tabs(number_of_tabs),
+      label(label),
+      creation_date(creation_date) {}
+
+AdNotificationTimingTaskLog::AdNotificationTimingTaskLog(
+    const AdNotificationTimingTaskLog& other) = default;
+
+AdNotificationTimingTaskLog::AdNotificationTimingTaskLog()
+    : id(0),
+      time(base::Time::Now()),
+      locale(std::string()),
+      number_of_tabs(0),
+      label(false),
+      creation_date(base::Time::Now()) {}
+
+AdNotificationTimingTaskLog::~AdNotificationTimingTaskLog() {}
+
+// AdNotificationTimingDataStore
+// -----------------------------------------------------
+
+AdNotificationTimingDataStore::AdNotificationTimingDataStore(
+    const base::FilePath& database_path)
+    : DataStore(database_path) {}
+
+bool AdNotificationTimingDataStore::Init(const int task_id,
+                                         const std::string& task_name,
+                                         const int max_number_of_records,
+                                         const int max_retention_days) {
+  return DataStore::Init(task_id, task_name, max_number_of_records,
```

Comment:
if you simply mark Init as virtual, you'll inherit the implementation from the base class